pkg/migration/migration.go

```diff
@@ -664,3 +712,23 @@ func (m *Service) RemoveOwnerReferenceFromSecrets() error {
 	klog.Infoln("...done")
 	return nil
 }
+
+// RetryOnError allows the caller to retry fn in case the error returned by fn is retriable
+// according to the provided function. backoff defines the maximum retries and the wait
+// interval between two retries.
+func RetryOnError(backoff wait.Backoff, fn func() error) error {
```

Comment:
I could have sworn we had a util function for this somewhere, but I can't seem to find it

Reply:
tomorrow, I will double recheck that. Basically,  this func was inspired by the client-go retry policy but maybe it is already somewhere in our code. 

btw. thanks for the review:)